packages/plugins/users-permissions/admin/src/hooks/useFetchRole/index.js

```diff
@@ -1,13 +1,14 @@
 import { useCallback, useReducer, useEffect, useRef } from 'react';
 import { useNotification } from '@strapi/helper-plugin';
+import { getFetchClient } from '@strapi/admin/admin/src/utils/getFetchClient';
```

Comment:
Yes, but if we want to share it across packages the helper-plugin is the way to do so, also because then the community can start using it. @joshuaellis do you have input here?

Reply:
I think deprecating the helper-plugin would mean creating a clearer new package – which we can't do in v4 for breaking changes & I don't believe `@strapi/plugin-user-permissions` has a dependency on `@strapi/admin` so this shouldn't be allowed imo.

Also lets be honest the import path – @strapi/admin/admin/src/utils/getFetchClient is a huge problem, no developer would remember this. IF we're insisting to keep it in `@strapi/admin` it needs to be exported from the root & each plugin needs to have `@strapi/admin` as a dependency.

Reply:
So we agree we have to move the hook into the helper plugin or does that require more discussion?

Reply:
I don't see another option. @simotae14 might have another option?

Reply:
Ok, I have understood your points @gu-stav and @joshuaellis ... I can move the implementation inside the helper-plugin at this point and postpone the discussion related to the deprecation of the helper-plugin.

Reply:
It looks like we have decided to deprecate the helper plugin eventually (even in v5), so why do we add the new APIs to it? I feel like we're making the future migration harder than it needs to be.

I personally prefer @joshuaellis's idea to surface the export more, like `import { getFetchClient } from @strapi/admin`. Also:

>each plugin needs to have @strapi/admin as a dependency

I think every plugin already has access to `@strapi/admin` since plugins are merged into the admin, without needing to add the dependency to the package.json

Reply:
> I think every plugin already has access to @strapi/admin since plugins are merged into the admin, without needing to add the dependency to the package.json

IMO this is poor practice. If you use a package in an app / plugin / other package you should declare it be explicit you're depending on it just being there which means if we change something e.g. we change the location of the hook then it could break the plugin using it for example.

Reply:
I agree. The helper-plugin is not deprecated yet, and we don't have a proper replacement. I'd actually argue it makes the migration easier because we don't start to mix two patterns which then have to be migrated to something else.

Reply:
Thanks for the details about why we would use the helper plugin.

@joshuaellis I agree with what you're saying about plugin dependencies, but putting everything in the package.json can also cause problems. We've had issues where plugins would use a different React version, and therefore the hooks would not work as there were 2 react copies.

Also we have the peerDependency system to specify which versions of Strapi a plugin is compatible with, which can help in case a new strapi version breaks a plugin

Reply:
>  We've had issues where plugins would use a different React version, and therefore the hooks would not work as there were 2 react copies.

This is because plugins should be treated as libraries therefore you don't build react in the bundle you transpile the plugin code and react is a peer-dep until the admin is built....

IMO this is highlighting (again) a massive flaw in our plugin system that needs to be rethinked in v5

> Also we have the peerDependency system to specify which versions of Strapi a plugin is compatible with, which can help in case a new strapi version breaks a plugin

This is good to know then. I think if you can export it from `@strapi/admin` then great (like in the suggestion, not the incredibly long path), but otherwise it needs to go in `@strapi/helper-plugin. Also keeping in mind that everything we let plugin devs use typically comes from the helper-plugin – this is a DX decision so i'll let @simotae14 make the call.

Reply:
Hello, finally we used the second solution, move the useFetchClient and getFetchClient code to the helper-plugin, because in the future removal of the helper plugin, won't be a big issue to move again these two to another place